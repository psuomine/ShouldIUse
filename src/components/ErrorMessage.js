import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { H3 } from 'components/styles/Heading'

const Message = styled(H3)`
  margin-top: 24px;
`

const ErrorMessage = ({ message }) => <Message color="error">{message}</Message>

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
}

export default ErrorMessage
