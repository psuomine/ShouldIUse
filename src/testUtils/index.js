import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import Theme from 'theme/Theme'

const customRender = (node, options) => render(<ThemeProvider theme={Theme}>{node}</ThemeProvider>, options)

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }
