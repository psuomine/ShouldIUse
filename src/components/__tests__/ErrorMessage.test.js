import React from 'react'
import { render } from 'testUtils'
import ErrorMessage from '../ErrorMessage'

describe('<ErrorMessage />', () => {
  it('renders error message based on props', () => {
    const message = 'Error happened!'
    const { getByText } = render(<ErrorMessage message={message} />)
    expect(getByText(message).textContent).toEqual(message)
  })

  it('snapshot', () => {
    const { container } = render(<ErrorMessage message="error happened" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
