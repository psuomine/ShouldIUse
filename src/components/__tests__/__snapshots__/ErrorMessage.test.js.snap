// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> snapshot 1`] = `
<h3
  class="sc-htpNat sc-bxivhb fvaKrV"
  color="error"
>
  error happened
</h3>
`;
